packages/runtime/src/components/hooks/useRouterLocaleSync.ts

```diff
@@ -1,16 +1,27 @@
 import { useRouter } from 'next/router'
-
+import { P, match } from 'ts-pattern'
 import { useEffect } from 'react'
+
 import { setLocale } from '../../state/actions'
 import { useDispatch } from '../../runtimes/react/hooks/use-dispatch'
 
+function getRouter() {
```

Comment:
Should we still name this `use*` since it's technically still a React hook and we want to make sure we follow the rules of hooks?

Reply:
Ah sure, one sec.